tests/terraform/graph/graph_builder/test_oci_policy.py

```diff
@@ -1,43 +1,38 @@
 from checkov.terraform.graph_manager import TerraformGraphManager
 
+@mock.patch.dict(os.environ, {"CKV_TERRAFORM_PROVIDER": "OCI"})
+def test_oci_policy_statements_with_provider_env_var():
+    # given
+    resources_dir = Path(__file__).parent.parent / "resources/oci_policies"
+    graph_manager = TerraformGraphManager(db_connector=NetworkxConnector())
+
+    # when
+    local_graph, _ = graph_manager.build_graph_from_source_directory(
+        source_dir=str(resources_dir), render_variables=True
+    )
+
+    # then
+    statements = local_graph.vertices[0].config["oci_identity_policy"]["example"]["statements"][0]
+    assert statements == [
+        "allow group group-admin-001 to use groups in tenancy where target.group.name != 'Administrators'"
+    ]
+
+
+def test_oci_policy_statements_without_provider_env_var():
+    # given
+    resources_dir = Path(__file__).parent.parent / "resources/oci_policies"
+    graph_manager = TerraformGraphManager(db_connector=NetworkxConnector())
+
+    # when
+    local_graph, _ = graph_manager.build_graph_from_source_directory(str(resources_dir), render_variables=True)
 
-class TestOCIPolicy(unittest.TestCase):
-    """This class tests OCI policy parsing and rendering."""
-
-    def test_oci_policy_statements(self):
-        """Test if OCI policy statements are evaluated correctly."""
-        current_dir = os.path.dirname(os.path.realpath(__file__))
-        resources_dir = os.path.join(current_dir, "..", "resources", "oci_policies")
-        graph_manager = TerraformGraphManager(db_connector=NetworkxConnector())
-
-        # When env variable PROVIDER is set to `OCI`, Checkov stops evaluating the `statements` attributes.
-        if os.environ.get("PROVIDER", "") == "OCI":
-            local_graph, _ = graph_manager.build_graph_from_source_directory(
-                resources_dir, render_variables=True
-            )
-            self.assertEqual(
-                str(
-                    local_graph.vertices[0].config["oci_identity_policy"]["example"][
-                        "statements"
-                    ]
-                ),
-                "[[\"allow group group-admin-001 to use groups in tenancy where target.group.name != 'Administrators'\"]]",
-            )
-        else:
-            local_graph, _ = graph_manager.build_graph_from_source_directory(
-                resources_dir, render_variables=True
-            )
-            self.assertNotEqual(
-                str(
-                    local_graph.vertices[0].config["oci_identity_policy"]["example"][
-                        "statements"
-                    ]
-                ),
-                "[[\"allow group group-admin-001 to use groups in tenancy where target.group.name != 'Administrators'\"]]",
-            )
+    # then
+    statements = local_graph.vertices[0].config["oci_identity_policy"]["example"]["statements"][0]
+    assert statements == ["True"]
```

Comment:
I can add a comment to clarify that `"True"` is a wrong value due to special syntax in OCI policy.